### PR TITLE
vimPlugins: treesitter-endwise, auto-hlsearch, update

### PR DESCRIPTION
**Pull request by caarlos0**
###### Description of changes

added some vim plugins:
- treesitter-endwise
- auto-hlsearch

by running:

```bash
pkgs/applications/editors/vim/plugins/update.py --add asiryk/auto-hlsearch.nvim --add RRethy/nvim-treesitter-endwise
```

which also updates the existing plugins, guess no harm done.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by linsui** [comment]
All mini.nvim plugins are already available in mini-nvim.

**Comment by caarlos0** [comment]
> All mini.nvim plugins are already available in mini-nvim.

oh, right, removing them

File: pkgs/applications/editors/vim/plugins/vim-plugin-names [hunk 1]
```diff
@@ -1196,3 +1197,4 @@ https://github.com/ziglang/zig.vim/,,
 https://github.com/mickael-menu/zk-nvim/,HEAD,
 https://github.com/troydm/zoomwintab.vim/,,
 https://github.com/nanotee/zoxide.vim/,,
+https://github.com/asiryk/auto-hlsearch.nvim/,,
```
Comment (figsoda) [inline pkgs/applications/editors/vim/plugins/vim-plugin-names:1200]: please keep this list sorted, the update script should do this automatically if you do this
```bash
pkgs/applications/editors/vim/plugins/update.py --add asiryk/auto-hlsearch.nvim --add RRethy/nvim-treesitter-endwise
```
then you can remove the update commits if you want

Comment (caarlos0) [reply pkgs/applications/editors/vim/plugins/vim-plugin-names]: done, thanks!

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 223493` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vimPlugins.sg-nvim</li>
  </ul>
</details>
<details>
  <summary>55 packages built:</summary>
  <ul>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.SpaceVim</li>
    <li>vimPlugins.auto-hlsearch-nvim</li>
    <li>vimPlugins.barbar-nvim</li>
    <li>vimPlugins.barbecue-nvim</li>
    <li>vimPlugins.ccc-nvim</li>
    <li>vimPlugins.cmp-copilot</li>
    <li>vimPlugins.compiler-explorer-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.copilot-lua</li>
    <li>vimPlugins.copilot-vim</li>
    <li>vimPlugins.dashboard-nvim</li>
    <li>vimPlugins.dial-nvim</li>
    <li>vimPlugins.firenvim</li>
    <li>vimPlugins.flutter-tools-nvim</li>
    <li>vimPlugins.glance-nvim</li>
    <li>vimPlugins.go-nvim</li>
    <li>vimPlugins.heirline-nvim</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.legendary-nvim</li>
    <li>vimPlugins.lsp-zero-nvim</li>
    <li>vimPlugins.lspsaga-nvim-original</li>
    <li>vimPlugins.mason-lspconfig-nvim</li>
    <li>vimPlugins.mason-nvim</li>
    <li>vimPlugins.mason-tool-installer-nvim</li>
    <li>vimPlugins.mini-nvim</li>
    <li>vimPlugins.neoconf-nvim</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neotest-haskell</li>
    <li>vimPlugins.nlsp-settings-nvim</li>
    <li>vimPlugins.null-ls-nvim</li>
    <li>vimPlugins.nvim-biscuits</li>
    <li>vimPlugins.nvim-cokeline</li>
    <li>vimPlugins.nvim-dap-ui</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-surround</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-endwise</li>
    <li>vimPlugins.nvim-web-devicons</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.package-info-nvim</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.telescope-file-browser-nvim</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vifm-vim</li>
    <li>vimPlugins.vim-abolish</li>
    <li>vimPlugins.vim-go</li>
    <li>vimPlugins.vim-rhubarb</li>
    <li>vimPlugins.vim-sensible</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vimtex</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**